aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -241,43 +292,70 @@ sub remove_and_create_vn_ars
         foreach my $t (@exisvnet) {
             my $arinfo = $t->get_ar(%ar_opts);
             if ($remove) {
-                # Detect Quattor and id first
-                $arid = $self->detect_vn_ar_quattor($arinfo);
-                if (defined($arid)) {
-                    $main::this_app->debug(1, "AR template to remove from $vnet: ", $ardata->{ar});
-                    my $rmid = $t->rmar($arid);
-                    if (defined($rmid)) {
-                        $main::this_app->info("Removed from vnet: $vnet AR id: $arid");
-                    } else {
-                        $main::this_app->error("Unable to remove AR id: $arid from vnet: $vnet");
-                    }
-                } else {
-                    $main::this_app->error("Quattor flag not found within AR. ", 
-                                        "ONE AII is not allowed to remove this AR.");
-                }
+                $self->remove_vn_ars($one, $arinfo, $vnet, $ardata, $t);
             } elsif (!$remove and $arinfo) {
-                # Update the AR info
-                $main::this_app->debug(1, "AR template to update from $vnet: ", $ardata->{ar});
-                $arid = $t->updatear($ardata->{ar});
-                if (defined($arid)) {
-                    $main::this_app->info("Updated $vnet AR id: ", $arid);
-                } else {
-                    $main::this_app->error("Unable to update AR from vnet: $vnet");
-                }
+                $self->update_vn_ars($one, $vnet, $ardata, $t);
             } else { 
-                # Create a new network AR
-                $main::this_app->debug(1, "New AR template in $vnet: ", $ardata->{ar});
-                $arid = $t->addar($ardata->{ar});
-                if (defined($arid)) {
-                    $main::this_app->info("Created new $vnet AR id: ", $arid);
-                } else {
-                    $main::this_app->error("Unable to create new AR within vnet: $vnet");
-                }
+                $self->create_vn_ars($one, $vnet, $ardata, $t);
             }
        }
     }
 }
 
+# Create VN AR leases
```

Comment:
@jrha virtual network address range acronym, yes I will include a comment to clarify this...